cpp/src/arrow/sparse_tensor.cc

```diff
@@ -55,6 +57,36 @@ class SparseTensorConverter {
   Status Convert() { return Status::Invalid("Unsupported sparse index"); }
 };
 
+// ----------------------------------------------------------------------
+// IncrementIndex for SparseCOOIndex and SparseCSFIndex
+
+void IncrementIndex(std::vector<int64_t>& coord, int64_t n, std::vector<int64_t> shape,
+                    int64_t ndim) {
```

Comment:
The type of `shape` should be `const std::vector<int64_t>&`, and `ndim` should be taken from `shape.size()`.

Reply:
It is better to make `n` const because `n` isn't changed in this function.

Reply:
Good point! Refactoring. Sorry I missed this.